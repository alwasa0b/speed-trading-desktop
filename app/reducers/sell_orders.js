import {
  UPDATE_SELL_ORDER_TYPE_TYPE,
  UPDATE_SELL_ORDER_PRICE_PRICE
} from "../constants/sell";

export default (state = {}, action) => {
  switch (action.type) {
    case UPDATE_SELL_ORDER_TYPE_TYPE:
      return {
        ...state,
        [action.symbol]: {
          ...state[action.symbol],
          order_type: action.order_type
        }
      };
    case UPDATE_SELL_ORDER_PRICE_PRICE:
      return {
        ...state,
        [action.symbol]: {
          ...state[action.symbol],
          price: action.price
        }
      };
    default:
      return state;
  }
};
