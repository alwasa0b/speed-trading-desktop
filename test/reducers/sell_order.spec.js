import sell_orders from "../../app/reducers/sell_orders";
import {
  update_sell_order_type,
  update_sell_order_price,
  update_quantity,
  update_quantity_type
} from "../../app/actions/sell";

describe("reducers", () => {
  describe("sell_order", () => {
    it("should handle initial state", () => {
      expect(sell_orders(undefined, {})).toMatchSnapshot();
    });

    it("should handle update_sell_order_type", () => {
      expect(
        sell_orders(
          undefined,
          update_sell_order_type({
            order_type: "test_symbol_type",
            symbol: "test_symbol"
          })
        )
      ).toMatchSnapshot();
    });

    it("should handle update_sell_order_price", () => {
      expect(
        sell_orders(
          undefined,
          update_sell_order_price({ price: 1, symbol: "test_symbol" })
        )
      ).toMatchSnapshot();
    });

    it("should handle update_quantity", () => {
      expect(
        sell_orders(
          undefined,
          update_quantity({ quantity: 1, symbol: "test_symbol" })
        )
      ).toMatchSnapshot();
    });

    it("should handle update_quantity_type", () => {
      expect(
        sell_orders(
          undefined,
          update_quantity_type({
            quantity_type: "test_type",
            symbol: "test_symbol"
          })
        )
      ).toMatchSnapshot();
    });

    it("should handle POSITIONS_UPDATED", () => {
      expect(
        sell_orders(undefined, {
          type: "POSITIONS_UPDATED",
          data: [
            { symbol: "symbol1", quantity: 1 },
            { symbol: "symbol2", quantity: 2 }
          ]
        })
      ).toMatchSnapshot();
    });

    it("should only update quantity_type and quantity for symbol1", () => {
      expect(
        sell_orders(
          {
            symbol1: {
              quantity: 1,
              quantity_type: "count"
            },
            symbol2: {
              quantity: 2,
              quantity_type: "count"
            }
          },
          update_quantity_type({
            quantity_type: "test_type",
            symbol: "symbol1"
          })
        )
      ).toMatchSnapshot();
    });
  });
});
