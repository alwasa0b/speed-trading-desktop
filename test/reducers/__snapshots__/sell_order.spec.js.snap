// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers sell_order should handle POSITIONS_UPDATED 1`] = `
Object {
  "symbol1": Object {
    "quantity": 1,
    "quantity_type": "count",
  },
  "symbol2": Object {
    "quantity": 2,
    "quantity_type": "count",
  },
}
`;

exports[`reducers sell_order should handle initial state 1`] = `Object {}`;

exports[`reducers sell_order should handle update_quantity 1`] = `
Object {
  "test_symbol": Object {
    "quantity": 1,
  },
}
`;

exports[`reducers sell_order should handle update_quantity_type 1`] = `
Object {
  "test_symbol": Object {
    "quantity": undefined,
    "quantity_type": "test_type",
  },
}
`;

exports[`reducers sell_order should handle update_sell_order_price 1`] = `
Object {
  "test_symbol": Object {
    "price": 1,
  },
}
`;

exports[`reducers sell_order should handle update_sell_order_type 1`] = `
Object {
  "test_symbol": Object {
    "order_type": "test_symbol_type",
  },
}
`;

exports[`reducers sell_order should only update quantity_type and quantity for symbol1 1`] = `
Object {
  "symbol1": Object {
    "quantity": undefined,
    "quantity_type": "test_type",
  },
  "symbol2": Object {
    "quantity": 2,
    "quantity_type": "count",
  },
}
`;
